### PR TITLE
Travis cleanup

### PR DESCRIPTION
**Pull request by furiel**
This PR is the continuation for https://github.com/balabit/syslog-ng/pull/2727

This patchset contains a little cleanup for .travis.yml

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
THIS IS AWESOME. thank you. It will make it much easier to improve travis and to integrate it with the jenkins instance.

I went through patch-by-patch, but didn't test it myself.

**Comment by alltilla** [comment]
I ran Travis on this code, and passed. We should rerun the Travis on this PR.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by furiel** [comment]
There is a little progress: the java11-java8 conflict was introduced when I moved the two refactored jobs from main level into the matrix. It turns out travis allocates trusty by default outside the matrix, and xenial when inside the matrix. For some reason travis-xenial vm contains java11. After adding `dist: trusty` those two jobs are passing.

**Comment by alltilla** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by alltilla** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: .travis.yml [hunk 1]
```diff
@@ -103,21 +103,15 @@ matrix:
...
-            make pytest-linters &&
-            make pytest-self-check &&
-            make pytest-check;
-          else
-            make V=1 install;
-            return $S;
-          fi;
+        - make --keep-going -j $(nproc) || make V=1 install;
+        - make install
+          . scripts/get-libjvm-path.sh || return 0;
```
Comment (Kokan) [inline .travis.yml]: The return should still use `$?`.

Comment (furiel) [reply .travis.yml]: Thanks. Fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**